### PR TITLE
Add generalisation

### PR DESCRIPTION
**Pull request by danji90**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,118 +20,97 @@
...
+    "stylelint-scss": "3.21.0"
   },
   "scripts": {
     "build": "react-scripts build",
     "deploy:ghpages": "gh-pages -d build",
     "docs": "./node_modules/.bin/jsdoc -c jsdoc.json",
     "eject": "react-scripts eject",
-    "format": "prettier --write 'src/**/*.js' && eslint 'src/**/*.js' --fix",
+    "format": "prettier --write 'src/**/*.{js,jsx}' && eslint 'src/**/*.{js,jsx}' --fix && stylelint 'src/**/*.css' 'src/**/*.scss' --fix",
     "lint": "eslint src/**/*.js src/**/*.jsx",
```
Comment (oterral) [inline package.json:78]: can you add also stylelint check here ?

File: package.json [hunk 2]
```diff
@@ -184,7 +163,8 @@
   },
   "prettier": {
     "trailingComma": "all",
-    "singleQuote": true
+    "singleQuote": true,
+    "arrowParens": "always"
```
Comment (oterral) [inline package.json:167]: not bad this one, we should use it everywhere

File: src/Components/FloorSwitcher/FloorSwitcher.jsx [hunk 3]
```diff
@@ -116,53 +114,53 @@ class FloorSwitcher extends PureComponent {
...
+  //       />
+  //       <Button
+  //         className={`tm-floor-button-mobile ${nextButtonClass}`}
+  //         onClick={() => nextFloor && this.selectFloor(nextFloor)}
+  //       >
+  //         <FiArrowDown />
+  //       </Button>
+  //     </div>
+  //   );
+  // }
```
Comment (oterral) [inline src/Components/FloorSwitcher/FloorSwitcher.jsx:163]: not needed anymore ?

File: src/Components/MapComponent/MapComponent.jsx [hunk 4]
```diff
@@ -44,21 +43,20 @@ import * as actions from '../../store/actions';
...
-const zoom = 6;
 let cbKey = null;
 
 /**
  * The only true map that shows inside the application.
  * @category Map
  */
 class MapComponent extends PureComponent {
-  static getExtentCenter = extent => {
+  static getExtentCenter = (extent) => {
```
Comment (oterral) [inline src/Components/MapComponent/MapComponent.jsx:59]: Don't hesitate to move static function inside an utils file. Make the code smaller and easier to test

**Review by oterral — APPROVED** [review]
PR looks good. I didn't test

File: src/utils.jsx [hunk 5]
```diff
@@ -28,11 +28,48 @@ export const findMotIcon = name => {
...
     c.toFixed(decimal),
   );
 };
 
-export const to3857 = coord => {
+export const to3857 = (coord) => {
   return transform(coord, 'EPSG:4326', 'EPSG:3857');
 };
+
+export const getGeneralization = (mot, zoom) => {
```
Comment (oterral) [inline src/utils.jsx:50]: maybe a test for this function?

File: src/config/styleConfig.js [hunk 6]
```diff
@@ -2,13 +2,12 @@
...
 
 // Convert '0.0' to '0'
-const cleanFloor = floor => {
-  return !isNaN(floor) ? parseFloat(floor, 10).toString() : floor;
-};
+const cleanFloor = (floor) =>
+  !isNaN(floor) ? parseFloat(floor, 10).toString() : floor;
 
-const lineStyler = lineStyle => {
-  return lineStyle.map(
```
Comment (oterral) [inline src/config/styleConfig.js:10]: is Prettier or eslint that have done that ? I prefer to keep the return . Maybe it's done in the next commit.

**Review by oterral — CHANGES_REQUESTED** [review]

File: src/Components/FloorSwitcher/FloorSwitcher.jsx (hunk 3 above)
Comment (danji90) [reply src/Components/FloorSwitcher/FloorSwitcher.jsx:163]: I have no idea, but isAppWidthSmallerThanS has been commented out for a while, so the whole render method is useless. I dunno, maybe we still need it in the future

File: src/Components/MapComponent/MapComponent.jsx (hunk 4 above)
Comment (danji90) [reply src/Components/MapComponent/MapComponent.jsx:59]: These static functions were all useless I just noticed, must be some relics from a while ago.
I nevertheless created a utils folder and moved separated some of the util functions into separate files

File: src/config/styleConfig.js (hunk 6 above)
Comment (danji90) [reply src/config/styleConfig.js:10]: No idea, but prettier can handle both variants

File: package.json (hunk 1 above)
Comment (danji90) [reply package.json:78]: done

**Review by oterral — APPROVED** [review]

**Pull request merged**